tests/core/test_simplicialcomplex.py

```diff
@@ -508,4 +508,88 @@ def test_freeze(edgelist1):
     with pytest.raises(XGIError):
         SC.remove_simplex_ids_from([0, 1])
 
-    assert SC.is_frozen()
+    assert SC.is_frozen
+
+
+def test_cleanup():
+    SC = xgi.SimplicialComplex()
+    SC.add_simplices_from([["a", "b", "c"], ["e", "f"]])
+    SC.add_nodes_from(["d", "g"])
+
+    assert set(SC.nodes) == {"a", "b", "c", "d", "e", "f", "g"}
+
+    # test removing isolates
+    cleanSC = SC.cleanup(connected=False, relabel=False, in_place=False)
+    assert set(cleanSC.nodes) == {"a", "b", "c", "e", "f"}
+    assert set(cleanSC.edges) == {0, 1, 2, 3, 4}
+    simplices = cleanSC.edges.members()
+    assert frozenset({"a", "b", "c"}) in simplices
+    assert frozenset({"e", "f"}) in simplices
+    assert frozenset({"a", "b"}) in simplices
+    assert frozenset({"a", "c"}) in simplices
+    assert frozenset({"b", "c"}) in simplices
+
+    # test getting giant component
+    cleanSC = SC.cleanup(isolates=True, relabel=False, in_place=False)
+    assert set(cleanSC.nodes) == {"a", "b", "c"}
+    assert cleanSC.num_edges == 4
+
+    # test relabel
+    cleanSC = SC.cleanup(isolates=True, connected=False, in_place=False)
+    assert set(cleanSC.nodes) == {0, 1, 2, 3, 4, 5, 6}
+    assert cleanSC.num_edges == 5
+    simplices = cleanSC.edges.members()
+    assert frozenset({0, 1, 2}) in simplices
+    assert frozenset({3, 4}) in simplices
+    assert frozenset({0, 1}) in simplices
+    assert frozenset({0, 2}) in simplices
+    assert frozenset({1, 2}) in simplices
+
+    ### In-place versions
+
+    # test removing isolates
+    cleanSC = SC.copy()
+    cleanSC.cleanup(connected=False, relabel=False)
+    assert set(cleanSC.nodes) == {"a", "b", "c", "e", "f"}
+    assert set(cleanSC.edges) == {0, 1, 2, 3, 4}
+    simplices = cleanSC.edges.members()
+    assert frozenset({"a", "b", "c"}) in simplices
+    assert frozenset({"e", "f"}) in simplices
+    assert frozenset({"a", "b"}) in simplices
+    assert frozenset({"a", "c"}) in simplices
+    assert frozenset({"b", "c"}) in simplices
+
+    # test getting giant component
+    cleanSC = SC.copy()
+    cleanSC.cleanup(isolates=False, relabel=False)
+    assert set(cleanSC.nodes) == {"a", "b", "c"}
+    assert cleanSC.num_edges == 4
+
+    # test relabel
+    cleanSC = SC.copy()
+    cleanSC.cleanup(connected=False)
+    assert set(cleanSC.nodes) == {0, 1, 2, 3, 4}
+    assert cleanSC.num_edges == 5
+    simplices = cleanSC.edges.members()
+    assert frozenset({0, 1, 2}) in simplices
+    assert frozenset({3, 4}) in simplices
+    assert frozenset({0, 1}) in simplices
+    assert frozenset({0, 2}) in simplices
+    assert frozenset({1, 2}) in simplices
+
+
+def test_remove_node(edgelist1):
```

Comment:
what's the difference with the test below? 
I thought the problem with #445 was that `H._node` should be `{}`, but I cannot see if we're testing that now?

Reply:
I test that `S._node` is the dual of `S._edge` below, but I will add an explicit test for the example in #445.